### PR TITLE
Add tests with empty iterable for asyncMap and asyncValMap

### PR DESCRIPTION
**Pull request by davidstrouk**

### CODE REVIEW

File: src/functional.test.js [hunk 1]
```diff
@@ -53,14 +53,26 @@ test("test asyncFirst all fail", async () => {
   expect(result).toBeFalsy();
 });
 
-test("test async map", async () => {
-  const result = await asyncMap(
-    (input) => Promise.resolve(multiply(input, 2)),
-    [1, 2, 3]
-  );
+describe("asyncMap", () => {
```
Comment (nitzo) [inline src/functional.test.js]: consider writing parametrized tests with jest

**Review by nitzo — APPROVED** [review]
